<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="HWC" />

<meta name="date" content="2025-07-25" />

<title>Myocarditis_Immpath</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Immpath_Myocarditis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Myocarditis_Immpath</h1>
<h4 class="author">HWC</h4>
<h4 class="date">2025-07-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Immpath_Myocarditis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250727code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250727)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250727code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250727)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/R/ImmPath_Myocarditis/data/Myocarditis_AMvsHealthy_merged_layers_filtered_25.05.25.rds
</td>
<td style="text-align:left;">
data/Myocarditis_AMvsHealthy_merged_layers_filtered_25.05.25.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHWC0406ImmPathMyocarditistree637e8eac89917228e303792858d54ed6cd21297etargetblank637e8eaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/HWC0406/ImmPath_Myocarditis/tree/637e8eac89917228e303792858d54ed6cd21297e" target="_blank">637e8ea</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomHWC0406ImmPathMyocarditistree637e8eac89917228e303792858d54ed6cd21297etargetblank637e8eaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/HWC0406/ImmPath_Myocarditis/tree/637e8eac89917228e303792858d54ed6cd21297e" target="_blank">637e8ea</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/ImmPath_Myocarditis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>##load merged file 
fileNam &lt;- &quot;/R/ImmPath_Myocarditis/data/Myocarditis_AMvsHealthy_merged_layers_filtered_25.05.25.rds&quot;
seuratM &lt;- readRDS(fileNam)
table(seuratM$dataset)</code></pre>
<pre><code>
     312581_05-5_20230308_Hu_nucseq_SG36_EMB_GEM      312581_06-6_20230308_Hu_nucseq_SG37_EMB_GEM 
                                            6327                                             1172 
   353921_28-28_20240529_Hu_nucseq_Immpath_EMB18    353921_31-31_20240529_Hu_nucseq_Immpath_EMB27 
                                             375                                              634 
   353921_32-32_20240529_Hu_nucseq_Immpath_EMB29    353921_34-34_20240529_Hu_nucseq_Immpath_EMB35 
                                             345                                             1296 
   353921_35-35_20240529_Hu_nucseq_Immpath_EMB37 359861_09-11_20240723_Hu_nucseq_Ber_Pat5_EM979_1 
                                             625                                              790 
359861_10-13_20240723_Hu_nucseq_Ber_Pat5_EM979_2  359861_11-1_20240730_Hu_nucseq_Ber_Pat1_EMB1537 
                                            1572                                              583 
 359861_12-2_20240730_Hu_nucseq_Ber_Pat2_EMB1041     359861_16-8_20240730_Hu_nucseq_Immpath_EMB47 
                                             972                                              583 
    359861_17-9_20240730_Hu_nucseq_Immpath_EMB48   377461_5-5_20250212_Hu_nucseq_USZ_Immpath_AM54 
                                             442                                              722 
  377461_6-6_20250212_Hu_nucseq_USZ_Immpath_AM59   377461_7-7_20250212_Hu_nucseq_USZ_Immpath_AM61 
                                             957                                             1102 
  377461_8-8_20250212_Hu_nucseq_USZ_Immpath_AM63      o27533_1_12-12_20220203_Hu_nucseq_EMB31_GEM 
                                            1454                                              244 
       o27936_1_7-7_20220309_Hu_nucseq_EMB32_GEM  o28576_1_01-1_20220525_Hu_nucseq_Graz_1_EMB_GEM 
                                            1285                                             3010 
 o28576_1_02-2_20220525_Hu_nucseq_Graz_2_EMB_GEM   o28576_1_08-8_20220525_Hu_nucseq_Graz_8_HH_GEM 
                                            1760                                             4005 
  o28576_1_09-9_20220525_Hu_nucseq_Graz_9_HH_GEM o28576_1_10-10_20220525_Hu_nucseq_Graz_10_HH_GEM 
                                            3052                                             3922 
o28576_1_11-11_20220525_Hu_nucseq_Graz_11_HH_GEM o28576_1_12-12_20220525_Hu_nucseq_Graz_12_HH_GEM 
                                            4265                                             3853 
    o292731_1-1_20220818_Hu_nucseq_Graz_9_HH_GEM    o292731_2-2_20220818_Hu_nucseq_Graz_13_HH_GEM 
                                            6434                                            11568 
    o292731_3-3_20220818_Hu_nucseq_SG_33_EMB_GEM   o294781_01-1_20220912_Hu_nucseq_Graz_21_HH_GEM 
                                            8414                                             1465 
  o294781_02-2_20220912_Hu_nucseq_Graz_22_HH_GEM   o294781_03-3_20220912_Hu_nucseq_Graz_23_HH_GEM 
                                            2064                                              866 
  o294781_04-4_20220912_Hu_nucseq_Graz_24_HH_GEM  o294781_06-6_20220912_Hu_nucseq_Graz_15_EMB_GEM 
                                            2181                                             4944 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14    15 
17713 12809 14117  9399  6655  6613  3652  2103  1883  2368   698  1074   689   718   484   497 
   16    17    18    19    20    21    22    23 
  357   368   432   141   178   141    98    96 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
83283 </code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-plots" class="section level2">
<h2>feature plots</h2>
<pre class="r"><code>FeaturePlot(seuratAM, features = &quot;ENSG00000125845.BMP2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratAM, features = &quot;ENSG00000125378.BMP4&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratAM, features = &quot;ENSG00000166923.GREM1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratAM, features = &quot;ENSG00000180875.GREM2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-2-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratAM, features = &quot;ENSG00000125845.BMP2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), order = TRUE)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-2-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratAM, features = &quot;ENSG00000125378.BMP4&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), order = TRUE)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-2-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratAM, features = &quot;ENSG00000166923.GREM1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), order = TRUE)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-2-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratAM, features = &quot;ENSG00000180875.GREM2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), order = TRUE)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-2-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="violin-plots" class="section level2">
<h2>Violin plots</h2>
<pre class="r"><code>Idents(seuratAM) &lt;- seuratAM$clusterName

VlnPlot(seuratAM, features = &quot;ENSG00000125845.BMP2&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#BE3144&quot;, &quot;#dfc27d&quot;))+
  stat_compare_means(method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratAM, features = &quot;ENSG00000125378.BMP4&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#BE3144&quot;, &quot;#dfc27d&quot;))+
  stat_compare_means(method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratAM, features = &quot;ENSG00000166923.GREM1&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#BE3144&quot;, &quot;#dfc27d&quot;))+
  stat_compare_means(method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratAM, features = &quot;ENSG00000180875.GREM2&quot;, pt.size = 0.01, group.by =&quot;pat_sub&quot;) +
  scale_fill_manual(values = c(&quot;#BE3144&quot;, &quot;#dfc27d&quot;))+
  stat_compare_means(method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratAM, features = &quot;ENSG00000125845.BMP2&quot;, pt.size = 0.01, group.by =&quot;clusterName&quot;)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratAM, features = &quot;ENSG00000125378.BMP4&quot;, pt.size = 0.01, group.by =&quot;clusterName&quot;)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-3-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratAM, features = &quot;ENSG00000166923.GREM1&quot;, pt.size = 0.01, group.by =&quot;clusterName&quot;)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-3-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(seuratAM, features = &quot;ENSG00000180875.GREM2&quot;, pt.size = 0.01, group.by =&quot;clusterName&quot;)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-3-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratAM) &lt;- seuratAM$pat_sub
markerGenes &lt;- FindAllMarkers(seuratAM, only.pos=T) %&gt;%   
dplyr::filter(p_val_adj &lt; 0.1)</code></pre>
</div>
<div id="relative-abundance-per-condition" class="section level1">
<h1>Relative abundance per condition</h1>
<pre><code>[1] 44729 39608
[1] 44729 43675</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="relative-abundance-per-sample" class="section level1">
<h1>Relative abundance per sample</h1>
<pre><code>[1] 44729  6327
[1] 44729  1172
[1] 44729   375
[1] 44729   634
[1] 44729   345
[1] 44729  1296
[1] 44729   625
[1] 44729   583
[1] 44729   442
[1] 44729   722
[1] 44729   957
[1] 44729  1102
[1] 44729  1454
[1] 44729   244
[1] 44729  1285
[1] 44729  3010
[1] 44729  1760
[1] 44729  8414
[1] 44729  4944
[1] 44729  2362
[1] 44729   583
[1] 44729   972
[1] 44729  4005
[1] 44729  3052
[1] 44729  3922
[1] 44729  4265
[1] 44729  3853
[1] 44729  6434
[1] 44729 11568
[1] 44729  1465
[1] 44729  2064
[1] 44729   866
[1] 44729  2181</code></pre>
<pre><code>                   Var1 Freq      percent           sample
ZH-Immpath-AM-10.1  CM1  925 0.1461988304 ZH-Immpath-AM-10
ZH-Immpath-AM-10.2  CM2  439 0.0693851746 ZH-Immpath-AM-10
ZH-Immpath-AM-10.3  CM3   47 0.0074284811 ZH-Immpath-AM-10
ZH-Immpath-AM-10.4  Fb1 1928 0.3047257784 ZH-Immpath-AM-10
ZH-Immpath-AM-10.5  Fb2    5 0.0007902639 ZH-Immpath-AM-10
ZH-Immpath-AM-10.6  Fb3   28 0.0044254781 ZH-Immpath-AM-10</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/abundance%20per%20sample-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/abundance%20per%20sample-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="total-cell-populations" class="section level3 tabset">
<h3 class="tabset">Total cell populations</h3>
</div>
<div id="t-cell-ratio-vs-bmp-pathway-genes" class="section level3">
<h3>T cell ratio vs BMP pathway genes</h3>
<pre class="r"><code>## Extract the BMP4 expression from the dataset
b&lt;-DotPlot(seuratAM,features = c(&quot;ENSG00000125378.BMP4&quot;, &quot;ENSG00000125845.BMP2&quot; , &quot;ENSG00000166923.GREM1&quot;, &quot;ENSG00000180875.GREM2&quot;), group.by=&quot;patient&quot;)
BMP4_path &lt;- as.data.frame(b$data)
colnames(BMP4_path)[colnames(BMP4_path) == &quot;id&quot;] &lt;- &quot;patient&quot;


## extract lymphocyte ratio from seurat object
datList &lt;- NULL
for(con in unique(seuratAM$patient)){
  seuratAMSub &lt;- subset(seuratAM, patient==con)
  print(dim(seuratAMSub))
  dat_con &lt;- as.data.frame(table(seuratAMSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratAMSub)) %&gt;% mutate(sample=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 44729  6327
[1] 44729  1172
[1] 44729   375
[1] 44729   634
[1] 44729   345
[1] 44729  1296
[1] 44729   625
[1] 44729   583
[1] 44729   442
[1] 44729   722
[1] 44729   957
[1] 44729  1102
[1] 44729  1454
[1] 44729   244
[1] 44729  1285
[1] 44729  3010
[1] 44729  1760
[1] 44729  8414
[1] 44729  4944
[1] 44729  2362
[1] 44729   583
[1] 44729   972
[1] 44729  4005
[1] 44729  3052
[1] 44729  3922
[1] 44729  4265
[1] 44729  3853
[1] 44729  6434
[1] 44729 11568
[1] 44729  1465
[1] 44729  2064
[1] 44729   866
[1] 44729  2181</code></pre>
<pre class="r"><code>dat_all_p &lt;- do.call(&quot;rbind&quot;, datList)
dat_tc &lt;- NULL
dat_tc &lt;- subset.data.frame(dat_all_p, Var1 == &quot;Lymphocytes&quot;) 
colnames(dat_tc)[colnames(dat_tc) == &quot;sample&quot;] &lt;- &quot;patient&quot;


# Option 1: Separate avg.exp per gene into wide format
BMP4_avgexp_wide &lt;- BMP4_path %&gt;%
  dplyr::select(patient, features.plot, avg.exp) %&gt;%
  tidyr::pivot_wider(names_from = features.plot, values_from = avg.exp)


# Option 2: Also pivot percent.exp if needed
BMP4_avgexp_scaled &lt;- BMP4_path %&gt;%
  dplyr::select(patient, features.plot, avg.exp.scaled) %&gt;%
  tidyr::pivot_wider(names_from = features.plot, values_from = avg.exp.scaled)

BMP4_wide &lt;- left_join(BMP4_avgexp_wide, BMP4_avgexp_scaled, by = &quot;patient&quot;, suffix = c(&quot;_avg&quot;, &quot;_avg.exp.scaled&quot;))

colnames(BMP4_wide) &lt;- sapply(colnames(BMP4_wide), function(x) {
  if (grepl(&quot;^ENSG[0-9]+\\.&quot;, x)) {
    sub(&quot;^ENSG[0-9]+\\.&quot;, &quot;&quot;, x)
  } else {
    x
  }
})

## merge different data frames
T_BMPpath &lt;- merge(dat_tc, BMP4_wide, by = &quot;patient&quot;)

T_BMPpath &lt;- T_BMPpath %&gt;%
  mutate(cond = ifelse(grepl(&quot;^healthy&quot;, patient, ignore.case = TRUE), &quot;Healthy&quot;, &quot;AM&quot;))</code></pre>
<div id="t-cell-ratio-vs-bmp-pathway-genes_average-expression"
class="section level4">
<h4>T cell ratio vs BMP pathway genes_average expression</h4>
<pre class="r"><code># List of gene expression columns to plot as y
geneset1 &lt;- c( &quot;BMP4_avg&quot;, &quot;BMP2_avg&quot;, &quot;GREM1_avg&quot;,&quot;GREM2_avg&quot;)

for (gene in geneset1) {
  p &lt;- ggplot(T_BMPpath, aes_string(x = &quot;percent&quot;, y = gene)) +  # aes_string for dynamic variable
    geom_point(color = &quot;orange&quot;) +
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = TRUE) +
    stat_cor(method = &quot;pearson&quot;, label.x = 0.05, label.y = max(T_BMPpath[[gene]], na.rm = TRUE)) +
    ggtitle(paste(&quot;T cell total -&quot;, gene)) +
    theme_classic()
  
  print(p)  # Show each plot
}</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="t-cell-ratio-vs-bmp-pathway-genes_average-expression-scaled"
class="section level4">
<h4>T cell ratio vs BMP pathway genes_average expression scaled</h4>
<pre class="r"><code># List of gene expression columns to plot as y
geneset2 &lt;- c(&quot;BMP4_avg.exp.scaled&quot;, &quot;BMP2_avg.exp.scaled&quot;, &quot;GREM1_avg.exp.scaled&quot;, &quot;GREM2_avg.exp.scaled&quot;)

for (gene in geneset2) {
  p &lt;- ggplot(T_BMPpath, aes_string(x = &quot;percent&quot;, y = gene)) +  # aes_string for dynamic variable
    geom_point(color = &quot;orange&quot;) +
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = TRUE) +
    stat_cor(method = &quot;pearson&quot;, label.x = 0.05, label.y = max(T_BMPpath[[gene]], na.rm = TRUE)) +
    ggtitle(paste(&quot;T cell total -&quot;, gene)) +
    theme_classic()
  
  print(p)  # Show each plot
}</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="stroma-lymphocytes" class="section level3 tabset">
<h3 class="tabset">Stroma &amp; Lymphocytes</h3>
<div id="t-cell-ratio-vs-bmp-pathway-genes---stroma-lymphocytes"
class="section level4">
<h4>T cell ratio vs BMP pathway genes - Stroma &amp; Lymphocytes</h4>
<pre class="r"><code>seuratAM_Stroma &lt;- subset(seuratAM, clusterName %in% c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC1&quot;, &quot;VSMC2&quot;,&quot;BEC1&quot;,&quot;BEC2&quot;, &quot;BEC3&quot;, &quot;LECs&quot;, &quot;Lymphocytes&quot;)) 
Idents(seuratAM_Stroma) &lt;- seuratAM_Stroma$clusterName
DimPlot(seuratAM_Stroma, reduction = &quot;umap&quot;,label = TRUE, repel = TRUE, cols = colclusterName, raster = T)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes-%20Stroma%20&%20Lymphocytes-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Extract the BMP4 expression from the dataset
e&lt;-DotPlot(seuratAM_Stroma,features = c(&quot;ENSG00000125378.BMP4&quot;, &quot;ENSG00000125845.BMP2&quot; , &quot;ENSG00000166923.GREM1&quot;, &quot;ENSG00000180875.GREM2&quot;), group.by=&quot;patient&quot;)
BMP_path_stroma &lt;- as.data.frame(e$data)
colnames(BMP_path_stroma)[colnames(BMP_path_stroma) == &quot;id&quot;] &lt;- &quot;patient&quot;

# Option 1: Separate avg.exp per gene into wide format
BMP_avgexp_stroma &lt;- BMP_path_stroma %&gt;%
  dplyr::select(patient, features.plot, avg.exp) %&gt;%
  tidyr::pivot_wider(names_from = features.plot, values_from = avg.exp)


# Option 2: Also pivot percent.exp if needed
BMP_avgexp_scaled_stroma &lt;- BMP_path_stroma %&gt;%
  dplyr::select(patient, features.plot, avg.exp.scaled) %&gt;%
  tidyr::pivot_wider(names_from = features.plot, values_from = avg.exp.scaled)

BMP_path_merStroma &lt;- left_join(BMP_avgexp_stroma, BMP_avgexp_scaled_stroma, by = &quot;patient&quot;, suffix = c(&quot;_avg&quot;, &quot;_avg.exp.scaled&quot;))

colnames(BMP_path_merStroma) &lt;- sapply(colnames(BMP_path_merStroma), function(x) {
  if (grepl(&quot;^ENSG[0-9]+\\.&quot;, x)) {
    sub(&quot;^ENSG[0-9]+\\.&quot;, &quot;&quot;, x)
  } else {
    x
  }
})

## merge different data frames
T_BMP_path_merStroma&lt;- merge(dat_tc, BMP_path_merStroma, by = &quot;patient&quot;)

T_BMP_path_merStroma &lt;- T_BMP_path_merStroma %&gt;%
  mutate(cond = ifelse(grepl(&quot;^healthy&quot;, patient, ignore.case = TRUE), &quot;Healthy&quot;, &quot;AM&quot;))</code></pre>
</div>
<div
id="t-cell-ratio-vs-bmp-pathway-genes_average-expression--stroma-lymphocytes"
class="section level4">
<h4>T cell ratio vs BMP pathway genes_average expression- Stroma &amp;
Lymphocytes</h4>
<pre class="r"><code># List of gene expression columns to plot as y
geneset1 &lt;- c( &quot;BMP4_avg&quot;, &quot;BMP2_avg&quot;, &quot;GREM1_avg&quot;,&quot;GREM2_avg&quot;)

for (gene in geneset1) {
  p &lt;- ggplot(T_BMP_path_merStroma, aes_string(x = &quot;percent&quot;, y = gene)) +  # aes_string for dynamic variable
    geom_point(color = &quot;orange&quot;) +
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = TRUE) +
    stat_cor(method = &quot;pearson&quot;, label.x = 0.05, label.y = max(T_BMP_path_merStroma[[gene]], na.rm = TRUE)) +
    ggtitle(paste(&quot;T cell total -&quot;, gene)) +
    theme_classic()
  
  print(p)  # Show each plot
}</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Stroma%20&%20Lymphocytes-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Stroma%20&%20Lymphocytes-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Stroma%20&%20Lymphocytes-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Stroma%20&%20Lymphocytes-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="t-cell-ratio-vs-bmp-pathway-genes_average-expression-scaled--stroma-lymphocytes"
class="section level4">
<h4>T cell ratio vs BMP pathway genes_average expression scaled- Stroma
&amp; Lymphocytes</h4>
<pre class="r"><code># List of gene expression columns to plot as y
geneset2 &lt;- c(&quot;BMP4_avg.exp.scaled&quot;, &quot;BMP2_avg.exp.scaled&quot;, &quot;GREM1_avg.exp.scaled&quot;, &quot;GREM2_avg.exp.scaled&quot;)

for (gene in geneset2) {
  p &lt;- ggplot(T_BMP_path_merStroma, aes_string(x = &quot;percent&quot;, y = gene)) +  # aes_string for dynamic variable
    geom_point(color = &quot;orange&quot;) +
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = TRUE) +
    stat_cor(method = &quot;pearson&quot;, label.x = 0.05, label.y = max(T_BMPpath[[gene]], na.rm = TRUE)) +
    ggtitle(paste(&quot;T cell total -&quot;, gene)) +
    theme_classic()
  
  print(p)  # Show each plot
}</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Stroma%20&%20Lymphocytes-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Stroma%20&%20Lymphocytes-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Stroma%20&%20Lymphocytes-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Stroma%20&%20Lymphocytes-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="fb-lymphocytes" class="section level3 tabset">
<h3 class="tabset">Fb &amp; Lymphocytes</h3>
<div id="t-cell-ratio-vs-bmp-pathway-genes---fb-lymphocytes"
class="section level4">
<h4>T cell ratio vs BMP pathway genes - Fb &amp; Lymphocytes</h4>
<pre class="r"><code>seuratAM_FBLym &lt;- subset(seuratAM, clusterName %in% c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC1&quot;, &quot;VSMC2&quot;,&quot;Lymphocytes&quot;)) 
Idents(seuratAM_FBLym) &lt;- seuratAM_FBLym$clusterName
DimPlot(seuratAM_FBLym, reduction = &quot;umap&quot;,label = TRUE, repel = TRUE, cols = colclusterName, raster = T)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes-%20Fb%20&%20Lymphocytes-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Extract the BMP4 expression from the dataset
d&lt;-DotPlot(seuratAM_FBLym,features = c(&quot;ENSG00000125378.BMP4&quot;, &quot;ENSG00000125845.BMP2&quot; , &quot;ENSG00000166923.GREM1&quot;, &quot;ENSG00000180875.GREM2&quot;), group.by=&quot;patient&quot;)
BMP_path_FBLym &lt;- as.data.frame(d$data)
colnames(BMP_path_FBLym)[colnames(BMP_path_FBLym) == &quot;id&quot;] &lt;- &quot;patient&quot;

# Option 1: Separate avg.exp per gene into wide format
BMP_avgexp_FBLym &lt;- BMP_path_FBLym %&gt;%
  dplyr::select(patient, features.plot, avg.exp) %&gt;%
  tidyr::pivot_wider(names_from = features.plot, values_from = avg.exp)


# Option 2: Also pivot percent.exp if needed
BMP_avgexp_scaled_FBLym &lt;- BMP_path_FBLym %&gt;%
  dplyr::select(patient, features.plot, avg.exp.scaled) %&gt;%
  tidyr::pivot_wider(names_from = features.plot, values_from = avg.exp.scaled)

BMP_path_merFbLym &lt;- left_join(BMP_avgexp_FBLym, BMP_avgexp_scaled_FBLym, by = &quot;patient&quot;, suffix = c(&quot;_avg&quot;, &quot;_avg.exp.scaled&quot;))

colnames(BMP_path_merFbLym) &lt;- sapply(colnames(BMP_path_merFbLym), function(x) {
  if (grepl(&quot;^ENSG[0-9]+\\.&quot;, x)) {
    sub(&quot;^ENSG[0-9]+\\.&quot;, &quot;&quot;, x)
  } else {
    x
  }
})

## merge different data frames
T_BMP_path_merFbLym&lt;- merge(dat_tc, BMP_path_merFbLym, by = &quot;patient&quot;)

T_BMP_path_merFbLym &lt;- T_BMP_path_merFbLym %&gt;%
  mutate(cond = ifelse(grepl(&quot;^healthy&quot;, patient, ignore.case = TRUE), &quot;Healthy&quot;, &quot;AM&quot;))</code></pre>
</div>
<div
id="t-cell-ratio-vs-bmp-pathway-genes_average-expression--fb-lymphocytes"
class="section level4">
<h4>T cell ratio vs BMP pathway genes_average expression- Fb &amp;
Lymphocytes</h4>
<pre class="r"><code># List of gene expression columns to plot as y
geneset1 &lt;- c( &quot;BMP4_avg&quot;, &quot;BMP2_avg&quot;, &quot;GREM1_avg&quot;,&quot;GREM2_avg&quot;)

for (gene in geneset1) {
  p &lt;- ggplot(T_BMP_path_merFbLym, aes_string(x = &quot;percent&quot;, y = gene)) +  # aes_string for dynamic variable
    geom_point(color = &quot;orange&quot;) +
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = TRUE) +
    stat_cor(method = &quot;pearson&quot;, label.x = 0.05, label.y = max(T_BMPpath[[gene]], na.rm = TRUE)) +
    ggtitle(paste(&quot;T cell total -&quot;, gene)) +
    theme_classic()
  
  print(p)  # Show each plot
}</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Fb%20&%20Lymphocytes-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Fb%20&%20Lymphocytes-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Fb%20&%20Lymphocytes-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Fb%20&%20Lymphocytes-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="t-cell-ratio-vs-bmp-pathway-genes_average-expression-scaled--fb-lymphocytes"
class="section level4">
<h4>T cell ratio vs BMP pathway genes_average expression scaled- Fb
&amp; Lymphocytes</h4>
<pre class="r"><code># List of gene expression columns to plot as y
geneset2 &lt;- c(&quot;BMP4_avg.exp.scaled&quot;, &quot;BMP2_avg.exp.scaled&quot;, &quot;GREM1_avg.exp.scaled&quot;, &quot;GREM2_avg.exp.scaled&quot;)

for (gene in geneset2) {
  p &lt;- ggplot(T_BMP_path_merFbLym, aes_string(x = &quot;percent&quot;, y = gene)) +  # aes_string for dynamic variable
    geom_point(color = &quot;orange&quot;) +
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = TRUE) +
    stat_cor(method = &quot;pearson&quot;, label.x = 0.05, label.y = max(T_BMPpath[[gene]], na.rm = TRUE)) +
    ggtitle(paste(&quot;T cell total -&quot;, gene)) +
    theme_classic()
  
  print(p)  # Show each plot
}</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Fb%20&%20Lymphocytes-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Fb%20&%20Lymphocytes-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Fb%20&%20Lymphocytes-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Fb%20&%20Lymphocytes-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="fb-only" class="section level3 tabset">
<h3 class="tabset">Fb only</h3>
<div id="t-cell-ratio-vs-bmp-pathway--fb-only" class="section level4">
<h4>T cell ratio vs BMP pathway- Fb only</h4>
<pre class="r"><code>seuratAM_FB &lt;- subset(seuratAM, clusterName %in% c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC1&quot;, &quot;VSMC2&quot;)) 
Idents(seuratAM_FB) &lt;- seuratAM_FB$clusterName
DimPlot(seuratAM_FB, reduction = &quot;umap&quot;,label = TRUE, repel = TRUE, cols = colclusterName, raster = T)</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes-%20Fb%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Extract the BMP4 expression from the dataset
a&lt;-DotPlot(seuratAM_FB,features = c(&quot;ENSG00000125378.BMP4&quot;, &quot;ENSG00000125845.BMP2&quot; , &quot;ENSG00000166923.GREM1&quot;, &quot;ENSG00000180875.GREM2&quot;), group.by=&quot;patient&quot;)
BMP_path_FB &lt;- as.data.frame(a$data)
colnames(BMP_path_FB)[colnames(BMP_path_FB) == &quot;id&quot;] &lt;- &quot;patient&quot;

# Option 1: Separate avg.exp per gene into wide format
BMP_avgexp_FB &lt;- BMP_path_FB %&gt;%
  dplyr::select(patient, features.plot, avg.exp) %&gt;%
  tidyr::pivot_wider(names_from = features.plot, values_from = avg.exp)


# Option 2: Also pivot percent.exp if needed
BMP_avgexp_scaled_FB &lt;- BMP_path_FB %&gt;%
  dplyr::select(patient, features.plot, avg.exp.scaled) %&gt;%
  tidyr::pivot_wider(names_from = features.plot, values_from = avg.exp.scaled)

BMP_path_merFb &lt;- left_join(BMP_avgexp_FB, BMP_avgexp_scaled_FB, by = &quot;patient&quot;, suffix = c(&quot;_avg&quot;, &quot;_avg.exp.scaled&quot;))

colnames(BMP_path_merFb) &lt;- sapply(colnames(BMP_path_merFb), function(x) {
  if (grepl(&quot;^ENSG[0-9]+\\.&quot;, x)) {
    sub(&quot;^ENSG[0-9]+\\.&quot;, &quot;&quot;, x)
  } else {
    x
  }
})

## merge different data frames
T_BMP_path_merFb&lt;- merge(dat_tc, BMP_path_merFb, by = &quot;patient&quot;)

T_BMP_path_merFb &lt;- T_BMP_path_merFb %&gt;%
  mutate(cond = ifelse(grepl(&quot;^healthy&quot;, patient, ignore.case = TRUE), &quot;Healthy&quot;, &quot;AM&quot;))</code></pre>
</div>
<div id="t-cell-ratio-vs-bmp-pathway-genes_average-expression--fb-only"
class="section level4">
<h4>T cell ratio vs BMP pathway genes_average expression- Fb only</h4>
<pre class="r"><code># List of gene expression columns to plot as y
geneset1 &lt;- c( &quot;BMP4_avg&quot;, &quot;BMP2_avg&quot;, &quot;GREM1_avg&quot;,&quot;GREM2_avg&quot;)

for (gene in geneset1) {
  p &lt;- ggplot(T_BMP_path_merFb, aes_string(x = &quot;percent&quot;, y = gene)) +  # aes_string for dynamic variable
    geom_point(color = &quot;orange&quot;) +
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = TRUE) +
    stat_cor(method = &quot;pearson&quot;, label.x = 0.05, label.y = max(T_BMPpath[[gene]], na.rm = TRUE)) +
    ggtitle(paste(&quot;T cell total -&quot;, gene)) +
    theme_classic()
  
  print(p)  # Show each plot
}</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Fb%20only-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Fb%20only-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Fb%20only-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression-%20Fb%20only-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="t-cell-ratio-vs-bmp-pathway-genes_average-expression-scaled--fb-only"
class="section level4">
<h4>T cell ratio vs BMP pathway genes_average expression scaled- Fb
only</h4>
<pre class="r"><code># List of gene expression columns to plot as y
geneset2 &lt;- c(&quot;BMP4_avg.exp.scaled&quot;, &quot;BMP2_avg.exp.scaled&quot;, &quot;GREM1_avg.exp.scaled&quot;, &quot;GREM2_avg.exp.scaled&quot;)

for (gene in geneset2) {
  p &lt;- ggplot(T_BMP_path_merFb, aes_string(x = &quot;percent&quot;, y = gene)) +  # aes_string for dynamic variable
    geom_point(color = &quot;orange&quot;) +
    geom_smooth(method = &quot;lm&quot;, color = &quot;black&quot;, se = TRUE) +
    stat_cor(method = &quot;pearson&quot;, label.x = 0.05, label.y = max(T_BMPpath[[gene]], na.rm = TRUE)) +
    ggtitle(paste(&quot;T cell total -&quot;, gene)) +
    theme_classic()
  
  print(p)  # Show each plot
}</code></pre>
<p><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Fb%20only-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Fb%20only-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Fb%20only-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/ImmPath_Myocarditis.Rmd/T%20cell%20ratio%20vs%20BMP%20pathway%20genes_average%20expression%20scaled-%20Fb%20only-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Sun Jul 27 15:56:37 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] writexl_1.5.4               scRepertoire_2.0.7          kableExtra_1.4.0           
 [4] knitr_1.50                  magrittr_2.0.3              RColorBrewer_1.1-3         
 [7] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[10] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
[13] enrichplot_1.24.4           msigdbr_24.1.0              org.Hs.eg.db_3.19.1        
[16] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[19] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[22] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[25] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[28] forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[31] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[34] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[37] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[40] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[43] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[46] pheatmap_1.0.13             ggpubr_0.6.1                ggplot2_3.5.2              
[49] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[52] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.11.0            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-6            rjson_0.2.23              evmix_2.12               
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.4           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  textshaping_1.0.1        
 [25] BiocNeighbors_1.22.0      stringdist_0.9.15         shadowtext_0.1.5         
 [28] uwot_0.2.3                curl_6.4.0                tidytree_0.4.6           
 [31] mime_0.12                 evaluate_1.0.4            ComplexHeatmap_2.20.0    
 [34] stringi_1.8.7             backports_1.5.0           gsl_2.1-8                
 [37] lmerTest_3.1-3            qqconf_1.3.2              httpuv_1.6.15            
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [49] git2r_0.36.2              corpcor_1.6.10            systemfonts_1.2.3        
 [52] reformulas_0.4.1          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.6                 
 [61] ggrepel_0.9.6             labeling_0.4.3            fANCOVA_0.6-1            
 [64] SparseArray_1.4.8         DESeq2_1.44.0             ranger_0.17.0            
 [67] DEoptimR_1.1-3-1          reticulate_1.42.0         hexbin_1.28.5            
 [70] zoo_1.8-14                XVector_0.44.0            ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             patchwork_1.3.1           caTools_1.18.3           
 [79] data.table_1.17.6         ggtree_3.12.0             quantreg_6.1             
 [82] R.oo_1.27.1               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] ggrastr_1.0.2             fastDummies_1.7.5         gridGraphics_0.5-1       
 [88] lazyeval_0.2.2            yaml_2.3.10               survival_3.6-4           
 [91] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [94] progressr_0.15.1          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] xml2_1.3.8                TMB_1.9.17                spatstat.univar_3.1-3    
[109] mathjaxr_1.8-0            EnvStats_3.1.0            evd_2.3-7.1              
[112] iNEXT_3.0.1               aplot_0.2.7               scatterplot3d_0.3-44     
[115] ape_5.8-1                 spatstat.sparse_3.1-0     xtable_1.8-4             
[118] car_3.1-3                 plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.18.0           
[124] beeswarm_0.4.0            broom_1.0.8               nlme_3.1-164             
[127] lambda.r_1.2.4            MatrixModels_0.5-4        assertthat_0.2.1         
[130] lme4_1.1-37               digest_0.6.36             numDeriv_2016.8-1.1      
[133] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[136] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.2.0        
[139] glue_1.7.0                cachem_1.1.0              polyclip_1.10-7          
[142] generics_0.1.4            Biostrings_2.72.1         ggalluvial_0.12.5        
[145] mvtnorm_1.3-3             presto_1.0.0              parallelly_1.45.0        
[148] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[151] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[154] pbapply_1.7-2             httr2_1.1.2               spam_2.11-1              
[157] gson_0.1.0                graphlayouts_1.2.2        gtools_3.9.5             
[160] ggsignif_0.6.4            RcppEigen_0.3.4.0.2       shiny_1.11.0             
[163] GenomeInfoDbData_1.2.12   glmmTMB_1.1.11            R.utils_2.13.0           
[166] memoise_2.0.1             rmarkdown_2.29            scales_1.4.0             
[169] R.methodsS3_1.8.2         svglite_2.2.1             future_1.58.0            
[172] RANN_2.6.1                Cairo_1.6-2               spatstat.data_3.1-6      
[175] rstudioapi_0.17.1         cluster_2.1.6             mutoss_0.1-13            
[178] spatstat.utils_3.1-4      hms_1.1.3                 fitdistrplus_1.2-3       
[181] cowplot_1.1.3             colorspace_2.1-1          ggdendro_0.2.0           
[184] rlang_1.1.4               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[187] xts_0.14.1                truncdist_1.0-2           dotCall64_1.2            
[190] shinydashboard_0.7.3      ggforce_0.5.0             laeken_0.5.3             
[193] mgcv_1.9-1                xfun_0.52                 e1071_1.7-16             
[196] TH.data_1.1-3             iterators_1.0.14          abind_1.4-8              
[199] GOSemSim_2.30.2           treeio_1.28.0             futile.options_1.0.1     
[202] bitops_1.0-9              Rdpack_2.6.4              promises_1.3.3           
[205] scatterpie_0.2.5          RSQLite_2.4.1             qvalue_2.36.0            
[208] sandwich_3.1-1            fgsea_1.30.0              DelayedArray_0.30.1      
[211] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.1           
[214] prettyunits_1.2.0         boot_1.3-30               beachmat_2.20.0          
[217] SparseM_1.84-2            listenv_0.9.1             Rcpp_1.0.13              
[220] edgeR_4.2.2               workflowr_1.7.1           BiocSingular_1.20.0      
[223] tensor_1.5.1              MASS_7.3-60.2             progress_1.2.3           
[226] BiocParallel_1.38.0       cubature_2.1.4            babelgene_22.9           
[229] spatstat.random_3.4-1     R6_2.6.1                  fastmap_1.2.0            
[232] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[235] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[238] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[241] nnet_7.3-19               gtable_0.3.6              KernSmooth_2.23-24       
[244] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[247] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.4-3   
[250] lifecycle_1.0.4           blme_1.0-6                S7_0.2.0                 
[253] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[256] VGAM_1.1-13               robustbase_0.99-4-1       spatstat.geom_3.4-1      
[259] sn_2.1.1                  ggfun_0.1.9               future.apply_1.20.0      
[262] bslib_0.9.0               pillar_1.10.2             gplots_3.2.0             
[265] pcaMethods_1.96.0         locfit_1.5-9.12           jsonlite_2.0.0           
[268] GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] writexl_1.5.4               scRepertoire_2.0.7          kableExtra_1.4.0           
 [4] knitr_1.50                  magrittr_2.0.3              RColorBrewer_1.1-3         
 [7] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[10] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
[13] enrichplot_1.24.4           msigdbr_24.1.0              org.Hs.eg.db_3.19.1        
[16] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[19] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[22] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[25] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[28] forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[31] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[34] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[37] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[40] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[43] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[46] pheatmap_1.0.13             ggpubr_0.6.1                ggplot2_3.5.2              
[49] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[52] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.11.0            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-6            rjson_0.2.23              evmix_2.12               
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.4           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  textshaping_1.0.1        
 [25] BiocNeighbors_1.22.0      stringdist_0.9.15         shadowtext_0.1.5         
 [28] uwot_0.2.3                curl_6.4.0                tidytree_0.4.6           
 [31] mime_0.12                 evaluate_1.0.4            ComplexHeatmap_2.20.0    
 [34] stringi_1.8.7             backports_1.5.0           gsl_2.1-8                
 [37] lmerTest_3.1-3            qqconf_1.3.2              httpuv_1.6.15            
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [49] git2r_0.36.2              corpcor_1.6.10            systemfonts_1.2.3        
 [52] reformulas_0.4.1          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.6                 
 [61] ggrepel_0.9.6             labeling_0.4.3            fANCOVA_0.6-1            
 [64] SparseArray_1.4.8         DESeq2_1.44.0             ranger_0.17.0            
 [67] DEoptimR_1.1-3-1          reticulate_1.42.0         hexbin_1.28.5            
 [70] zoo_1.8-14                XVector_0.44.0            ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             patchwork_1.3.1           caTools_1.18.3           
 [79] data.table_1.17.6         ggtree_3.12.0             quantreg_6.1             
 [82] R.oo_1.27.1               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] ggrastr_1.0.2             fastDummies_1.7.5         gridGraphics_0.5-1       
 [88] lazyeval_0.2.2            yaml_2.3.10               survival_3.6-4           
 [91] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [94] progressr_0.15.1          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] xml2_1.3.8                TMB_1.9.17                spatstat.univar_3.1-3    
[109] mathjaxr_1.8-0            EnvStats_3.1.0            evd_2.3-7.1              
[112] iNEXT_3.0.1               aplot_0.2.7               scatterplot3d_0.3-44     
[115] ape_5.8-1                 spatstat.sparse_3.1-0     xtable_1.8-4             
[118] car_3.1-3                 plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.18.0           
[124] beeswarm_0.4.0            broom_1.0.8               nlme_3.1-164             
[127] lambda.r_1.2.4            MatrixModels_0.5-4        assertthat_0.2.1         
[130] lme4_1.1-37               digest_0.6.36             numDeriv_2016.8-1.1      
[133] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[136] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.2.0        
[139] glue_1.7.0                cachem_1.1.0              polyclip_1.10-7          
[142] generics_0.1.4            Biostrings_2.72.1         ggalluvial_0.12.5        
[145] mvtnorm_1.3-3             presto_1.0.0              parallelly_1.45.0        
[148] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[151] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[154] pbapply_1.7-2             httr2_1.1.2               spam_2.11-1              
[157] gson_0.1.0                graphlayouts_1.2.2        gtools_3.9.5             
[160] ggsignif_0.6.4            RcppEigen_0.3.4.0.2       shiny_1.11.0             
[163] GenomeInfoDbData_1.2.12   glmmTMB_1.1.11            R.utils_2.13.0           
[166] memoise_2.0.1             rmarkdown_2.29            scales_1.4.0             
[169] R.methodsS3_1.8.2         svglite_2.2.1             future_1.58.0            
[172] RANN_2.6.1                Cairo_1.6-2               spatstat.data_3.1-6      
[175] rstudioapi_0.17.1         cluster_2.1.6             mutoss_0.1-13            
[178] spatstat.utils_3.1-4      hms_1.1.3                 fitdistrplus_1.2-3       
[181] cowplot_1.1.3             colorspace_2.1-1          ggdendro_0.2.0           
[184] rlang_1.1.4               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[187] xts_0.14.1                truncdist_1.0-2           dotCall64_1.2            
[190] shinydashboard_0.7.3      ggforce_0.5.0             laeken_0.5.3             
[193] mgcv_1.9-1                xfun_0.52                 e1071_1.7-16             
[196] TH.data_1.1-3             iterators_1.0.14          abind_1.4-8              
[199] GOSemSim_2.30.2           treeio_1.28.0             futile.options_1.0.1     
[202] bitops_1.0-9              Rdpack_2.6.4              promises_1.3.3           
[205] scatterpie_0.2.5          RSQLite_2.4.1             qvalue_2.36.0            
[208] sandwich_3.1-1            fgsea_1.30.0              DelayedArray_0.30.1      
[211] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.1           
[214] prettyunits_1.2.0         boot_1.3-30               beachmat_2.20.0          
[217] SparseM_1.84-2            listenv_0.9.1             Rcpp_1.0.13              
[220] edgeR_4.2.2               workflowr_1.7.1           BiocSingular_1.20.0      
[223] tensor_1.5.1              MASS_7.3-60.2             progress_1.2.3           
[226] BiocParallel_1.38.0       cubature_2.1.4            babelgene_22.9           
[229] spatstat.random_3.4-1     R6_2.6.1                  fastmap_1.2.0            
[232] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[235] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[238] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[241] nnet_7.3-19               gtable_0.3.6              KernSmooth_2.23-24       
[244] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[247] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.4-3   
[250] lifecycle_1.0.4           blme_1.0-6                S7_0.2.0                 
[253] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[256] VGAM_1.1-13               robustbase_0.99-4-1       spatstat.geom_3.4-1      
[259] sn_2.1.1                  ggfun_0.1.9               future.apply_1.20.0      
[262] bslib_0.9.0               pillar_1.10.2             gplots_3.2.0             
[265] pcaMethods_1.96.0         locfit_1.5-9.12           jsonlite_2.0.0           
[268] GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
